File: Python/0001-0050/21. Merge Two Sorted Lists.py
# https://leetcode.com/problems/merge-two-sorted-lists/description/

class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        mergedNode = ListNode() # create new node
        head = mergedNode

        while list1 != None and list2 != None :
            if(list1.val <= list2.val) :
                mergedNode.next = list1
                list1 = list1.next
            else :
                mergedNode.next = list2
                list2 = list2.next 
            
            mergedNode = mergedNode.next
        

        if list1!=None:
            mergedNode.next = list1
        

        if list2!=None:
            mergedNode.next = list2
        

        return head.next
        

File: Java/0001-0050/21. Merge Two Sorted Lists.java
// https://leetcode.com/problems/merge-two-sorted-lists/description/

class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode mergedNode = new ListNode(); //create new node
        ListNode head = mergedNode;

        while(list1 != null && list2 != null){
            if(list1.val <= list2.val){
                mergedNode.next = list1;
                list1 = list1.next;
            }else{
                mergedNode.next = list2;
                list2 = list2.next; 
            }
            mergedNode = mergedNode.next;
        }

        if(list1!=null){
            mergedNode.next = list1;
        }

        if(list2!=null){
            mergedNode.next = list2;
        }

        return head.next;
    }
}
